src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'blue' | 'green';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
 
-}
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(
+    public color: Color,
+    ...args : number[]
+  ) {
+    if (args.some((side) => side <= 0)) {
+      throw new Error('One side cannot be bigger than other two!');
+    }
 
-export class Circle {
+    const longestSide = Math.max(...args);
 
+    if (longestSide >= args
+      .filter((side) => side !== longestSide)
+      .reduce((sum, side) => sum + side)) {
+      throw new Error('Side value cannot be lesser than 1!');
+    }
+
+    [this.a, this.b, this.c] = args;
+  }
+
+  getArea(): number {
+    const semiperimeter = 0.5 * (this.a + this.b + this.c);
+
+    return Number(Math
+      .sqrt(semiperimeter
+        * (semiperimeter - this.a)
+        * (semiperimeter - this.b)
+        * (semiperimeter - this.c)).toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = Shape.Circle;
+
+  constructor(public color: Color, public radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius value cannot be lesser than 1!');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor((Math.PI * (this.radius ** 2)) * 100) / 100;
```

Comment:
I rounded in this way all figures, not only Circle because of this condition, but I'm not sure if it's necessary. toFixed() it's like Math.round(), not Math.floor
![image](https://user-images.githubusercontent.com/87998640/229290801-abb68fde-b1c2-44e9-976c-06cc9b7bbd3a.png)